### ISSUE TITLE
clang -Wconstant-conversion conditional operator false positive

### ISSUE THREAD

**Issue by osandov** [llvm/llvm-project#64741]
The following C source code:

```c
int x = 0 ? 9223372036854775807L : 2147483648;
```

Produces the following warning when compiled with Clang:

```console
$ clang --version
clang version 16.0.6 (Fedora 16.0.6-2.fc38)
Target: x86_64-redhat-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
$ clang -Wconstant-conversion -fsyntax-only test.c
test.c:1:13: warning: implicit conversion from 'long' to 'int' changes value from 9223372036854775807 to -1 [-Wconstant-conversion]
int x = 0 ? 9223372036854775807L : 2147483647;
    ~       ^~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

`0 ? 9223372036854775807L : 2147483647` evaluates to `2147483647L`, so this warning is incorrect. GCC's similar `-Woverflow` doesn't trigger for this test case.

**Comment by shafik** [comment]
This does feel a bit noisy, wdyt @AaronBallman

**Comment by nathanchance** [comment]
I suspect this is https://github.com/llvm/llvm-project/issues/38137, based on https://godbolt.org/z/c9E5bPGhP. The Linux kernel has hit this in a couple of places (https://github.com/ClangBuiltLinux/linux/issues/92).

**Comment by AaronBallman** [comment]
@nathanchance is correct, this is a duplicate of #38137 (and boils down to trying to gin up a CFG at global scope), but also worth noting that `-Woverflow` in GCC also misses the overflow in the third operand: https://godbolt.org/z/fc98PPY76
